archunit/src/test/java/com/tngtech/archunit/library/plantuml/rules/JavaClassDiagramAssociationTest.java

```diff
@@ -57,19 +56,6 @@ public void get_target_package_identifiers_of_class() {
                 .containsOnly(expectedTarget1, expectedTarget2);
     }
 
-    @Test
-    public void rejects_class_not_contained_in_any_component() {
-        JavaClassDiagramAssociation javaClassDiagramAssociation = createAssociation(TestDiagram.in(temporaryFolder)
-                .component("SomeComponent").withStereoTypes("..someStereotype.")
-                .write());
-        JavaClass classNotContained = importClassWithContext(Object.class);
-
-        thrown.expect(IllegalStateException.class);
```

Comment:
We could keep the test case, but document the new behavior that a `NoSuchElementException` is thrown.